1.	Identify-Basic-Internet-Problems-with-ping.md

```diff
@@ -0,0 +1,113 @@
+
+**The "ping" command:**
+
+The ping command is a simple way to verify that another computer can receive information from you. The original author, Mike Muuss, actually named the program after the "ping" sound that a submarine sends to detect objects in the water. If an echo of the ping comes back, it means that there is something out there. In fact, "ping" uses a protocol named "Internet Control Message Protocol Echo Request" as part of its underlying software design.
+
+In its simplest form, the ping command provide two valuable pieces of information, whether the message was echoed back ("64 bytes from…") and how long it takes to receive the message back (e.g., time="6.396 ms"). Depending on what type of computer you are using, you may even get a summary containing minimum, maximum, average, and more. The response time is shown in "ms", or millisecond, which is 1/1000th of a second. A response time of 10ms or less is pretty fast, however values are often in the 100ms range. At much above 200ms you’ll probably notice that you have a sluggish connection.
+
+**When all is well:**
+
+This is what my ping response looks like on my Mac OS X computer when everything is working normally here in Malaysia:
+
+```
+MacBook-Pro:~ ajm$ ping Google.com
+PING google.com (216.58.196.46): 56 data bytes
+64 bytes from 216.58.196.46: icmp\_seq=0 ttl=55 time=6.396 ms
+64 bytes from 216.58.196.46: icmp\_seq=1 ttl=55 time=6.368 ms
+64 bytes from 216.58.196.46: icmp\_seq=2 ttl=55 time=26.773 ms
+64 bytes from 216.58.196.46: icmp\_seq=3 ttl=55 time=6.984 ms
+^C
+--- google.com ping statistics ---
+4 packets transmitted, 4 packets received, 0.0% packet loss
+round-trip min/avg/max/stddev = 6.368/11.630/26.773/8.746 ms
+```
+
+This is what my ping response looks like on a Windows computer when everything is working well:
+
+```
+C:\Users\BJM>ping Google.com
+Pinging google.com [216.58.196.46] with 32 bytes of data:
+Reply from 216.58.196.46: bytes=32 time=6ms TTL=128
+Reply from 216.58.196.46: bytes=32 time=15ms TTL=128
+Reply from 216.58.196.46: bytes=32 time=6ms TTL=128
+Reply from 216.58.196.46: bytes=32 time=6ms TTL=128
+Ping statistics for 216.58.196.46:
+    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
+Approximate round trip times in milli-seconds:
+    Minimum = 6ms, Maximum = 15ms, Average = 8ms
+```
+
+You can see from these examples that the connection is pretty good with average response times under 10ms.
+
+**When something is wrong (three examples):**
+
+So what would happen if I could not connect to Google.com? For example #1, I simulate a broken network connection to my Mac by unplugging my router from the wall, and re-run the command. The first thing I notice is that it takes a \*lot\* longer for the command to respond:
```

Comment:
> if I could not connect to ~~Google.com~~`Google.com`

Inline style, and **NOT** link, because you passed it as argument to some utility function in terminal.